pkg/controllers/provisioning/scheduling/topologygroup.go

```diff
@@ -245,9 +258,9 @@ func (t *TopologyGroup) nextDomainTopologySpread(pod *corev1.Pod, podDomains, no
 	}
 	if minDomain == "" {
 		// avoids an error message about 'zone in [""]', preferring 'zone in []'
-		return scheduling.NewRequirement(podDomains.Key, corev1.NodeSelectorOpDoesNotExist)
```

Comment:
Odd that we were using `podDomains` in the first place here, I thought there may have been a reason but it does look like `podDomains.Key` is always equal to `t.Key`. I do think using the topology group's key makes more sense.